pkg/allocateip/allocateip.go

```diff
@@ -416,16 +607,23 @@ func determineEnabledPoolCIDRs(node api.Node, ipPoolList api.IPPoolList, vxlan b
 			continue
 		}
 
-		// Check if desired encap is enabled in the IP pool, the IP pool is not disabled, and it is IPv4 pool since we don't support encap with IPv6.
-		if vxlan {
+		// Check if desired encap is enabled in the IP pool, the IP pool is not disabled, and it is IPv4 pool since we
+		// don't support encap with IPv6.
+		switch attrType {
+		case ipam.AttributeTypeVXLAN:
 			if (ipPool.Spec.VXLANMode == api.VXLANModeAlways || ipPool.Spec.VXLANMode == api.VXLANModeCrossSubnet) && !ipPool.Spec.Disabled && poolCidr.Version() == 4 {
 				cidrs = append(cidrs, *poolCidr)
 			}
-		} else {
+		case ipam.AttributeTypeIPIP:
 			// Check if IPIP is enabled in the IP pool, the IP pool is not disabled, and it is IPv4 pool since we don't support IPIP with IPv6.
 			if (ipPool.Spec.IPIPMode == api.IPIPModeCrossSubnet || ipPool.Spec.IPIPMode == api.IPIPModeAlways) && !ipPool.Spec.Disabled && poolCidr.Version() == 4 {
 				cidrs = append(cidrs, *poolCidr)
 			}
+		case ipam.AttributeTypeWireguard:
+			// Wireguard does not require a specific encap configuration on the pool.
```

Comment:
I think we should respect the IP pool's nodeSelector field though, right? I think you should double check that the IPAM code handles that properly when explicit IP pools are passed in. Otherwise we might need to do a selector match here.

Reply:
Yes we should - the node selector check is done above ^^^ so I think this is correct as it stands.

That said, it's also done in the IPAM code, so I could simplify this logic and remove the pool selection code entirely (since IPAM checks a pool is enabled and matches the node selector) and just return nil pools to allow the IPAM code to select pools for itself.  Having the logic here though seems less confusing.